File: 55. Jump Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        vector <int> T (nums.size());
        T[0] = nums[0];
        if (nums.size() > 1 && T[0] < 1)
            return false;
        for (int i = 0; i < nums.size(); i++)
        {
            T[i] = T[i-1] > nums[i] + i ? T[i-1] : nums[i] + i;
            if (nums.size() > i + 1 && T[i] < i + 1)
                return false;
        }
        return true;
    }
};

File: 55. Jump Game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        T = [0 for i in xrange(len(nums))]
        T[0] = nums[0]
        if len(nums) > 1 and T[0] < 1:
            return False
        for i in xrange(1, len(nums)):
            T[i] = max(T[i-1], nums[i] + i)
            if len(nums) > i+1 and T[i] < i+1:
                return False
        return True
